### PR TITLE
Feature/redirect after comment login

### PR DESCRIPTION
**Pull request by freedompraise**
## Description

This PR resolves an issue on the blog site that comes up when a user that's not logged in tries to comment on a blog post as seen on issue #323 
The changes made would login a user, and redirect to the comments-section. 
It also addresses #290 

## Changes Made

- [ ] Added new feature
- [x] Fixed bug
- [ ] Refactored code
- [ ] Other (please describe):

## Test Plan

I ran the tests on the READme

## Checklist

- [x] I have read the [contributing guidelines](link_to_contributing_guidelines)
- [ ] I have added tests to cover my changes and they all pass in addition to the existing tests.
- [ ] I have added documentation for my changes (if appropriate)

### CODE REVIEW

File: django_project/users/utils.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+def handle_no_permission(request, slug):
+    """
+    Handles the case when the user is not logged in and tries to access the view.
+    Redirects the user to the login page, and after successful login, redirects
+    back to the original page with the appropriate slug to submit the comment.
+    """
+    login_url = f"/login/?next=/post/{slug}/" + "#comments-section"
+
+    if not request.user.is_authenticated:
+        return redirect(login_url)
```
[the remark this answers is not in the document]
Comment (jsolly) [reply django_project/users/utils.py:13]: You don't need to use a plus sign if you're using an f-string

```py
login_url = f"/login/?next=/post/{slug}/#comments-section"
```

File: django_project/blog/templates/blog/comment/add_comment.html
Comment (jsolly) [inline django_project/blog/templates/blog/comment/add_comment.html:1]: What is the purpose of this change? If the user is on the post detail page, the post will surely have a slug.

I removed this code, and everything still worked the same.

File: django_project/users/utils.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    # If the user is logged in, redirect to the original page to submit the comment
+    return redirect("post-detail", slug=slug)
```
Comment (jsolly) [inline django_project/users/utils.py:16]: In what scenario would this function be called if the user was logged in? I believe all logged-in users can comment.

I am just not seeing the purpose of this last return statement. Maybe it needs to return a 500 or something>

If we decide to keep this, please improve the comment to something like this:

> "If the user is logged in, but doesn't have permission to comment, redirect to the original post"

**Review by jsolly — CHANGES_REQUESTED** [review]
Hey @freedompraise, this looks great! I like the way you used a handlle_no_permission function in users/utils.py.

Just left a couple of small comments. Once they are addressed, we can merge.

File: django_project/users/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (freedompraise) [reply django_project/users/utils.py:13]: oh. Noted

File: django_project/users/utils.py (hunk 2 above)
Comment (freedompraise) [reply django_project/users/utils.py]: I observed there's no need for this, so I took it off

File: django_project/blog/templates/blog/comment/add_comment.html
Comment (freedompraise) [reply django_project/blog/templates/blog/comment/add_comment.html:1]: Oh. I think I was trying to test something there. Shouldn't have committed it

**Review by jsolly — APPROVED** [review]

**Comment by jsolly** [comment]
Great work!
